Backend/Gmfctn/Controllers/AchievementController.cs

```diff
@@ -80,14 +95,21 @@ public async Task<ActionResult> CreateAchievement(AchievementCreateDTO Achieveme
                 throw Exc;
             }
         }
-        [HttpDelete("{id}")]
+        [HttpDelete("{Id}")]
+        [Authorize(Roles = "Admin")]
         public async Task<ActionResult> DeleteAchievement(Guid Id, CancellationToken Cancel)
         {
             try
             {
-                await UnitOfWork.AchievementRepository.Delete(Id, Cancel);
-                await UnitOfWork.SaveChangesAsync(Cancel);
-                return Ok();
+                if (await AchievementService.DeleteAchievement(Id, Cancel))
+                {
+                    return Ok();
+                }
+                return NotFound();
```

Comment:
The same as above about handling errors and return from Service logic

Reply:
Fixed